src/models/userHelpers.js

```diff
@@ -638,43 +647,50 @@ async function getExistingUsers(d2, options = {}) {
     });
     return users;
 }
-function addItems(items1, items2, shouldAdd) {
+
+function addItems(items1, items2, shouldAdd, updateStrategy) {
     if (!shouldAdd) {
         return items1;
     } else {
-        return _(items1)
-            .unionBy(items2, element => element.id)
-            .value();
+        return updateStrategy === "merge"
+            ? _(items1)
+                  .unionBy(items2, element => element.id)
+                  .value()
+            : items2;
     }
 }
 
-function getPayload(parentUser, destUsers, fields) {
+function getPayload(parentUser, destUsers, fields, updateStrategy) {
```

Comment:
[minor, no need to change] When a function starts growing, a typical refactor is to use a single options argument object. This way, the call to the function has each field named, which helps readability.